tests/unit/test_ccldap.py

```diff
@@ -0,0 +1,147 @@
+# -------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -------------------------------------------------------------------------
+
+import unittest
+
+from mockldap import MockLdap
+
+from libcodechecker.libauth import cc_ldap
+
+
+class CCLDAPTest(unittest.TestCase):
+
+    top = ('o=test', {'o': ['test']})
+    example = ('ou=example,o=test', {'ou': ['example']})
+    other = ('ou=other,o=test', {'ou': ['other']})
+    service_user = ('cn=service_user,ou=example,o=test',
+                    {'cn': ['service_user'], 'userPassword': ['servicepw']})
+    user2 = ('cn=user2,ou=other,o=test',
+             {'cn': ['user2'], 'userPassword': ['user2pw']})
+
+    # This is the content of our mock LDAP directory.
+    # It takes the form {dn: {attr: [value, ...], ...}, ...}.
+    directory = dict([top, example, other, service_user, user2])
+
+    # service_user is used as a service user in the configuration.
+    ldap_config = {"connection_url": "ldap://localhost/",
+                   "username": "cn=service_user,ou=example,o=test",
+                   "password": "servicepw",
+                   "referrals": False,
+                   "deref": "always",
+                   "accountBase": "ou=other,o=test",
+                   "accountScope": "subtree",
+                   "accountPattern": "(cn=$USN$)",
+                   "groupBase": "o=test",
+                   "groupScope": "subtree",
+                   "groupPattern": "",
+                   "groupNameAttr": ""
+                   }
+
+    @classmethod
+    def setup_class(cls):
+        cls.mockldap = MockLdap(cls.directory)
+
+    def setUp(self):
+        # Patch ldap.initialize
+        self.mockldap.start()
+        self.ldapobj = self.mockldap['ldap://localhost/']
+
+    def tearDown(self):
+        # Stop patching ldap.initialize and reset state.
+        self.mockldap.stop()
+        del self.ldapobj
```

Comment:
Is this for explicit garbage collection triggering?

Reply:
Yes it's there so each test runs with a clean state.
For each test it will be initialized:
```
def setUp(self):
    # Patch ldap.initialize
    self.mockldap.start()
    self.ldapobj = self.mockldap['ldap://localhost/']
```
It could be set on class level, so all the tests use the same mock, but I think it is cleaner this way.